ansible_collections/arista/avd/roles/eos_designs/templates/overlay/ibgp/router-bgp.j2

```diff
@@ -3,7 +3,11 @@ router_bgp:
 {% if switch.evpn_role == "server" or switch.mpls_overlay_role == "server" %}
   bgp_cluster_id: {{ switch.bgp_cluster_id | arista.avd.default(switch.router_id) }}
 {% endif %}
-
+{% if switch.mpls_vpn_gateway is arista.avd.defined(true) and switch.mpls_vpn_domain_dpath_enabled is arista.avd.defined(true) %}
```

Comment:
all these double checks could be simplified if you create a single fact that has all the needed logic done already.